x-pack/solutions/security/plugins/security_solution_ess/public/navigation/navigation_tree.ts

```diff
@@ -65,6 +66,14 @@ export const createNavigationTree = (services: Services): NavigationTreeDefiniti
         {
           breadcrumbStatus: 'hidden',
           children: [
+            ...(services.uiSettings.get(SECURITY_SOLUTION_ENABLE_ASSET_INVENTORY_SETTING)
+              ? [
+                  {
+                    id: SecurityPageName.assetInventory,
+                    link: securityLink(SecurityPageName.assetInventory),
+                  },
+                ]
+              : []),
```

Comment:
I think this `uiSettings` check is not necessary:
```suggestion
            {
              id: SecurityPageName.assetInventory,
              link: securityLink(SecurityPageName.assetInventory),
            },
```

The link configuration for `SecurityPageName.assetInventory` is already defined with `uiSettingRequired`:

https://github.com/elastic/kibana/blob/bac0c982f555977bf360aaebd4948385ea669cd2/x-pack/solutions/security/plugins/security_solution/public/asset_inventory/links.ts#L19-L31

And, from what I see, `ENABLE_ASSET_INVENTORY_SETTING` and `SECURITY_SOLUTION_ENABLE_ASSET_INVENTORY_SETTING` have the same value. So, this link is already being filtered out when this UI setting is disabled.

The navigation tree only defines the link hierarchy statically, no conditional logic should be needed.

Reply:
Thanks a lot for reviewing this @semd, I didn't know it was all connected. I simplified the code thanks to your comment and also removed the tests since it should be standard behavior then